tests/integrationv2/test_sslyze.py

```diff
@@ -0,0 +1,251 @@
+        ])
+
+    server = managed_process(S2N, server_options, timeout=30)
+
+    scan_results = run_sslyze_scan(HOST, port, [scan_command])
+
+    for result in scan_results:
+        assert_scan_result_completed(result)
+
+        scan_results = result.scan_result
+        scan_attempts = get_scan_attempts(scan_results)
+        for scan_attempt in scan_attempts:
+            validate_scan_result(scan_attempt, protocol)
+
+    server.kill()
+
+
+def validate_certificate_scan_result(scan_attempt, protocol, certificate):
+    assert_scan_attempt_completed(scan_attempt)
+    result = scan_attempt.result
+
+    def validate_cipher_scan(result, protocol, certificate):
+        assert result.is_tls_version_supported is True
+
+        rejected_ciphers = [
+            cipher for rejected_cipher in result.rejected_cipher_suites
+            if (cipher := Ciphers.from_iana(rejected_cipher.cipher_suite.name))
+        ]
+
+        for cipher in rejected_ciphers:
+            # if a cipher is rejected, it should be an invalid test parameter in combination with the
+            # protocol/provider/cert
+            assert invalid_test_parameters(
+                protocol=protocol,
+                provider=S2N,
+                certificate=certificate,
+                cipher=cipher
+            )
+
+    def validate_curves_scan(result, protocol, certificate):
+        assert result.supports_ecdh_key_exchange is True
+
+        rejected_curves = [
+            curve for rejected_curve in result.rejected_curves
+            if (curve := {
+                "X25519": Curves.X25519,
+                "prime256v1": Curves.P256,
+                "prime384v1": Curves.P384,
+                "prime521v1": Curves.P521
+            }.get(rejected_curve.name))
+        ]
+
+        for curve in rejected_curves:
+            assert invalid_test_parameters(
+                protocol=protocol,
+                provider=S2N,
+                certificate=certificate,
+                curve=curve
+            )
+
+    {
+        sslyze.CipherSuitesScanResult: validate_cipher_scan,
+        sslyze.SupportedEllipticCurvesScanResult: validate_curves_scan,
+    }.get(type(result))(result, protocol, certificate)
+
+
+@pytest.mark.parametrize("protocol", PROTOCOLS_TO_TEST, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", CERTS_TO_TEST, ids=get_parameter_name)
+def test_sslyze_certificate_scans(managed_process, protocol, certificate):
+    port = next(available_ports)
+
+    server_options = ProviderOptions(
+        mode=S2N.ServerMode,
+        host=HOST,
+        port=port,
+        protocol=protocol,
+        key=certificate.key,
+        cert=certificate.cert,
+        insecure=True,
+        extra_flags=["--parallelize"]
+    )
+    server = managed_process(S2N, server_options, timeout=30)
+
+    scans = [CIPHER_SUITE_SCANS.get(protocol.value)]
+
+    # sslyze doesn't like ECDSA certs for curve scan
```

Comment:
Not sure what you mean by "doesn't like". Do you mean sslyze doesn't accept ECDSA certs?

Reply:
Yes, I think it might not accept them. I get a `BUG_IN_SSLYZE` error when trying to run the scan on only these certs.

Reply:
I think I would phrase this comment as "Ssslyze errors with ECDSA certificates for their curve scan" then. If we want to dig deeper we can, but it's probably out of scope for this pr.

Reply:
Fixed in 4428cc4017342ea0615b3493dc091a8df711bc46